### PR TITLE
IDB manager and reader

### PR DESCRIPTION
**Pull request by nicHoch**
a IDB manager for different versions in the file system
and a IDB reader

### CODE REVIEW

**Review by samaloney — CHANGES_REQUESTED** [review]
This looks pretty good few minor overall comments.

The doc strings are not consistent, I know it not fun but it would be good to get them sorted at this stage some examples can be found  [here](https://numpydoc.readthedocs.io/en/latest/format.html) and [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)

The test data files are a bit large I'm surprized they got past the pre-commit hook, I need to check that,  I've already created some code to download test data in #6 we might be able to adapt that as we are only going to have more and more test data.

I think the test failure maybe python version related what version are you running locally?

File: stixcore/idb/idb.py [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+
+        returns
+        ----------
+        True | False
+        """
+        if self.cur:
+            return True
+        return False
+
+    def reload(self, filename):
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/idb.py:53]: removed

File: stixcore/idb/idb.py [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+            'select PCF.PCF_DESCR, PLF.PLF_OFFBY, PLF.PLF_OFFBI, PCF.PCF_NAME,'
+            ' PCF.PCF_WIDTH, PCF.PCF_PFC,PCF.PCF_PTC, PCF.PCF_CURTX'
+            ' from PLF   inner join PCF  on PLF.PLF_NAME = PCF.PCF_NAME '
+            ' and PLF.PLF_SPID=? order by PLF.PLF_OFFBY asc')
+        args = (spid, )
+        res = self._execute(sql, args, 'dict')
+        self.parameter_structures[spid] = res
+        return res
+
+    def get_telecommand_info(self, header):
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/idb.py:364]: done

File: stixcore/idb/manager.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        """
+        Creates the manager for a given data path root
+
+        Parameters
+        ----------
+        data_path : `str` or `pathlib.Path`
+            Path to the directory with all IDB versions
+
+        """
+        self.data_root = data_root
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/manager.py:27]: have i used the wrong property notation: 
self.data_root is the property 
self._data_root the private variable

File: stixcore/idb/manager.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        raise ValueError(f'Version "{version_label}" not found in: "{self._get_filename_for_version(version_label)}"')
+
+if __name__ == '__main__': # pragma: no cover
+    a = IdbManager("./stixcore/idb/tests/data")
+    print(a.data_root)
+    versions = a.get_versions()
+    print(versions)
+    for v in versions :
+        print(v, a.has_version(v['version']))
+    print(a.get_idb('2.26.3'))
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/manager.py:164]: no

File: stixcore/idb/tests/test_idb.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    os.chdir(file_dir.parent / 'data')
+
+
+def teardown_function():
+    os.chdir(orig_directory)
+
+
+@pytest.fixture
+def idb():
+    return IdbManager("./").get_idb(VERSION)
```
[the remark this answers is not in the document]
Comment (nicHoch) [reply stixcore/idb/tests/test_idb.py:30]: works

**Comment by nicHoch** [comment]
revoked for PR https://github.com/i4Ds/STIXCore/pull/10

**Pull request closed without merging**